### PR TITLE
Monorepo

### PR DESCRIPTION
**Pull request by ForbesLindesay**
As described in #2675.  This combines many of pug’s separate packages into a single monorepo and replaces the various test runners with jest.  Although a few years ago, jest was pretty terrible, it’s recently become awesome.  It has out of the box support for:

 - **snapshots** - these have great diff views, and an easy mechanism for updating them when they change, letting us discard lots of our custom code for snapshotting the output of pug.
- **watch mode** - this completely changes the experience of using tests when authoring code.
- **coverage** - there is no need for awkward hacks or a separate tool, just pass `—coverage`.
- **monorepos** - because jest finds all files ending in `.test.js`, it works perfectly in a monorepo.  It saves us from needing to treat each package separately.
- **performance** - jest runs all the different files in parallel, meaning that running all the tests is pretty much as fast as just running the existing tests for only the main pug module.

### CODE REVIEW

File: packages/pug-attrs/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+language: node_js
```
Comment (TimothyGu) [inline packages/pug-attrs/.travis.yml]: These individual `.travis.yml`s can now be removed.

**Comment by thejameskyle** [comment]
Dunno if you already saw this but lerna also has a `lerna import <path/to/repo>` command for maintaining git history while merging repos in.

**Comment by ForbesLindesay** [comment]
Yes, the command was erroring on my local machine, and I decided I didn't really want all that history anyway.  The code has gone through so many big refactorings that the low level detail of all those individual commits is not very useful.

**Comment by ORESoftware** [comment]
Hope this doesn't turn out to be a nightmare! mono-repo ftw!

**Pull request merged**